### DISCUSSION TITLE
Filter by application, organization and role

### DISCUSSION THREAD

**Discussion by narayanpromax** [logto-io/logto#6097] [Q&A]
Does it support filtering users by role, application id and organization id?

I need a user list paginated that i sortable as per name and created_at and filtered by name, email and phone as well as permanent filters of ole, application id and organization id to be shown in frontend

**Comment by darcyYe** [comment]
We currently does not support search users by filtering with role/app/organization. But we have APIs that can do this:
1. For searching users with specific roles, you can checkout [`GET /api/roles/{id}/users` API](https://openapi.logto.io/operation/operation-get-api-roles-parameter-users).
2. Filtering user by app, you can get a list of users and then filter by appId by writing little code.
3. By getting all users in an org, you can refer to [this `GET /api/organizations/{id}/users` API](https://openapi.logto.io/operation/operation-get-api-organizations-parameter-users).

**Reply by IRediTOTO** [reply]
how about filter in org by role? I really need this

**Comment by narayanpromax** [comment]
I want APIs only any chance we can combine these filtering?

Like

api/organizations/:orgid/app/:appid/roles/:roleid/users
